lms/djangoapps/bulk_email/tasks.py

```diff
@@ -97,25 +97,31 @@
 )
 
 
-def _get_course_email_context(course):
+def _get_course_email_context(course, is_secure):
     """
     Returns context arguments to apply to all emails, independent of recipient.
+
+    Inputs are:
+    * `course`: Course related to the current email
+    * `is_secure`: Set to True to return https URLs, False to use http.
     """
     course_id = course.id.to_deprecated_string()
     course_title = course.display_name
     course_end_date = get_default_time_display(course.end)
-    course_url = 'https://{}{}'.format(
-        settings.SITE_NAME,
+    scheme = u'https' if is_secure else u'http'
```

Comment:
@bradenmacdonald Ach bummer..  I wasted a few hours redoing this fix using `request.is_secure()` 😦 .  But I agree, it is simpler to use `settings.HTTPS`.  Will ask you first next time 😄 , and update our documentation.